r/R/array.R

```diff
@@ -188,6 +188,11 @@ Array$create <- function(x, type = NULL) {
     return(out)
   }
 
+  if (inherits(x, "POSIXct") && attr(x, "tzone") == "") {
+    attr(x, "tzone") <- Sys.timezone()
```

Comment:
This is a bit tricky since a `"tzone"` attribute is not a must for all POSIXct objects. For example, if the object is created with `strptime()` it will not have a `"tzone"` attribute.

Reply:
And if there is no `"tzone"`` attribute, is it then a truly timezone-naive object?

Reply:
This is the reason for the second half of the `any()` statement:
```r
 if (inherits(x, "POSIXct") && any(attr(x, "tzone") == "", is.null(attr(x, "tzone"))))
```

Reply:
I will mark this as resolved.

Reply:
@jorisvandenbossche & @jonkeane  I'm unsolving this - not sure how to interpret the missing-ness of the `"tzone"` attribute (to respond to your other question @jorisvandenbossche).

Reply:
It looks to me like it's still an unmarked, but still definitely zoned timestamp:

```
> a <- strptime("2020-01-01", format = "%Y-%m-%d") + 1
> a
[1] "2020-01-01 00:00:01 CST"
> as.integer(a)
[1] 1577858401
```

If it was truly naive it should (1) not display the timezone in the print method) and (2) converting it to an integer is undefined (since you don't know how to go from the timestamp details to the epoch

Reply:
I came to the same conclusion as @jonkeane when I checked this earlier today: it still displays with a timezone, and also the underlying integer is the same compared to creating the POSIXct with tzone=""